sycl/test-e2e/GroupInterface/work_group.cpp

```diff
@@ -0,0 +1,146 @@
+#if defined(__cpp_lib_mdspan)
+      acc &= (wi.extents().rank() == 1);
+      acc &= (wi.extent(0) == 1);
+#endif
+      acc &= (wi.size() == 1);
+    });
+  });
+  q.submit([&](handler &h) {
+    sycl::accessor<bool, 0> acc{result, h};
+    h.parallel_for(nd_range<2>{range<2>{sz, sz}, range<2>{sz, sz}},
+                   [=](nd_item<2> item) {
+                     group<2> g = item.get_group();
+
+                     khr::work_group<2> wg = g;
+                     acc &= (wg.id() == g.get_group_id());
+                     acc &= (wg.linear_id() == g.get_group_linear_id());
+                     acc &= (wg.range() == g.get_group_range());
+#if defined(__cpp_lib_mdspan)
+                     acc &= (wg.extents().rank() == 2);
+                     acc &= (wg.extent(0) == g.get_local_range()[0]);
+                     acc &= (wg.extent(1) == g.get_local_range()[1]);
+#endif
+                     acc &= (wg.size() == g.get_local_linear_range());
+
+                     khr::member_item wi = get_member_item(wg);
+                     acc &= (wi.id() == g.get_local_id());
+                     acc &= (wi.linear_id() == g.get_local_linear_id());
+                     acc &= (wi.range() == g.get_local_range());
+#if defined(__cpp_lib_mdspan)
+                     acc &= (wi.extents().rank() == 2);
+                     acc &= (wi.extent(0) == 1);
+                     acc &= (wi.extent(1) == 1);
+#endif
+                     acc &= (wi.size() == 1);
+                   });
+  });
+  q.submit([&](handler &h) {
+    sycl::accessor<bool, 0> acc{result, h};
+    h.parallel_for(nd_range<3>{range<3>{sz, sz, sz}, range<3>{sz, sz, sz}},
+                   [=](nd_item<3> item) {
+                     group<3> g = item.get_group();
+
+                     khr::work_group<3> wg = g;
+                     acc &= (wg.id() == g.get_group_id());
+                     acc &= (wg.linear_id() == g.get_group_linear_id());
+                     acc &= (wg.range() == g.get_group_range());
+#if defined(__cpp_lib_mdspan)
+                     acc &= (wg.extents().rank() == 3);
+                     acc &= (wg.extent(0) == g.get_local_range()[0]);
+                     acc &= (wg.extent(1) == g.get_local_range()[1]);
+                     acc &= (wg.extent(2) == g.get_local_range()[2]);
+#endif
+                     acc &= (wg.size() == g.get_local_linear_range());
+
+                     khr::member_item wi = get_member_item(wg);
+                     acc &= (wi.id() == g.get_local_id());
+                     acc &= (wi.linear_id() == g.get_local_linear_id());
+                     acc &= (wi.range() == g.get_local_range());
+#if defined(__cpp_lib_mdspan)
+                     acc &= (wi.extents().rank() == 3);
+                     acc &= (wi.extent(0) == 1);
+                     acc &= (wi.extent(1) == 1);
+                     acc &= (wi.extent(2) == 1);
+#endif
+                     acc &= (wi.size() == 1);
+                   });
+  });
+  q.wait();
+
```

Comment:
nit, to consider.
I would like to suggest to split this (still in this file, but three RUN: with some param or at least separate status check for every nd_range<x>, now we have only 1 bool value that signals that something failed. For future analysis would be better if we have result a little bit more differentiated.